test/acceptance/features/steps/steps.py

```diff
@@ -104,9 +110,10 @@ def imported_nodejs_app_is_running(context, application_name):
         application.is_running(wait=True) | should.be_truthy.desc("Application is running")
     print("Nodejs application is running!!!")
     application.get_db_name_from_api() | should_not.be_none
-    context.nodejs_app_original_generation = application.get_observed_generation()
-    context.nodejs_app_original_pod_name = application.get_running_pod_name()
-    context.nodejs_app = application
+    context.application_original_generation = application.get_observed_generation()
+    context.application_original_pod_name = application.get_running_pod_name()
+    context.application = application
+    context.application_type = "nodejs"
```

Comment:
why do we need to know the application type? Cannot not we make that NodeJs adn Quarkus implement the same API? I would to the later, because the number of applications will increase over the time.

Reply:
I feel it is easy handle with type, moreover there are rev as a resource that we check for knative, otherwise for others it checks pods.

Reply:
removed the application type, and handling this by capturing get_generation and the same will be validated in is_redeployed()
thanks